gnxi_tester/config/history_test.go

```diff
@@ -30,6 +30,158 @@ type Want struct {
 	err     error
 }
 
+func TestSetTarget(t *testing.T) {
+	viper.SetConfigFile("/tmp/config.yml")
+	dir, err := os.Getwd()
+	if err != nil {
+		t.Errorf("Error getting working directory: %v", err)
+	}
+	certPath := path.Join(dir, "ca.crt")
+	certKeyPath := path.Join(dir, "ca.key")
+	tests := []struct {
+		name          string
+		targetName    string
+		targetAddress string
+		targetCA      string
+		targetCAKey   string
+		config        map[string]Target
+		lastTarget    string
+		want          Want
+	}{
+		{
+			name: "No targets in history, no target specified",
+			want: Want{
+				err:     errors.New("No targets in history and no target specified"),
+				targets: map[string]Target{},
+			},
+		},
+		{
+			name: "No target specified",
+			config: map[string]Target{"myhost.com": {
+				Address: "localhost:9339",
+				Ca:      certPath,
+				CaKey:   certKeyPath,
+			}},
+			lastTarget: "myhost.com",
+			want: Want{
+				err: nil,
+				targets: map[string]Target{"myhost.com": {
+					Address: "localhost:9339",
+					Ca:      certPath,
+					CaKey:   certKeyPath,
+				}},
+			},
+		},
+		{
+			name: "Non-existent target",
+			config: map[string]Target{"myhost.com": {
+				Address: "localhost:9339",
+				Ca:      certPath,
+				CaKey:   certKeyPath,
+			}},
+			targetName: "nonexistenttarget",
+			lastTarget: "myhost.com",
+			want: Want{
+				err: errors.New("Target not found"),
+				targets: map[string]Target{"myhost.com": {
+					Address: "localhost:9339",
+					Ca:      certPath,
+					CaKey:   certKeyPath,
+				}},
+			},
+		},
+		{
+			name:          "Add new target",
+			targetName:    "myhost.com",
+			targetAddress: "localhost:9339",
+			targetCA:      certPath,
+			targetCAKey:   certKeyPath,
+			want: Want{
+				targets: map[string]Target{
+					"myhost.com": {
+						Address: "localhost:9339",
+						Ca:      certPath,
+						CaKey:   certKeyPath,
+					},
```

Comment:
this very same content is repeated multiple times, save considerable amount of code lines by having it defined once and reused.

Reply:
Agreed! Will do!